src/main/webapp/app/overview/course-exercises/course-exercises.component.ts

```diff
@@ -60,15 +84,68 @@ export class CourseExercisesComponent implements OnInit, OnDestroy {
         this.paramSubscription.unsubscribe();
     }
 
-    public groupExercises(selectedOrder: number): void {
+    get filterType() {
+        return ExerciseFilter;
+    }
+
+    /**
+     * Return the total number of exercises for the logged in user
+     */
+    getNrOfExercises(): number {
+        return sum([...this.exerciseCountMap.values()]);
+    }
+
+    /**
+     * Reorders all displayed exercises
+     */
+    flipOrder() {
+        this.sortingOrder = this.sortingOrder === this.ASC ? this.DESC : this.ASC;
+        localStorage.setItem(SortFilterStorageKey.ORDER, this.sortingOrder.toString());
+        this.applyFiltersAndOrder();
+    }
+
+    /**
+     * Filters all displayed exercises by applying the selected activeFilters
+     * @param filters The filters which should be applied
+     * @param active Should the activeFilters be active or inactive
+     */
+    filterUpdate(filters: ExerciseFilter[]) {
+        filters.forEach(filter => (this.activeFilters.has(filter) ? this.activeFilters.delete(filter) : this.activeFilters.add(filter)));
```

Comment:
Because of the use of a mutable Map, it would be nicer to use a for loop here, just to make it obvious that the Map is being mutated here.

Reply:
Also I would call the method 'toggleFilters'.

Reply:
Renamed it